### ISSUE TITLE
Problem in logic when working with the count_instead_of_period parameter

### ISSUE THREAD

**Issue by Vikontrol** [nixys/nxs-backup#36]
```
# nxs-backup --version
nxs-backup 3.3.1
```

with count_instead_of_period enabled and the 24-1-0 scheme:

```
    retention:
      count_instead_of_period: true
      days: 24
      weeks: 1
      months: 0
```

On Sunday at the end of the day, there were 23 broken symbolic links in daily and one working one for the only backup in weekly.

**Comment by Vikontrol** [comment]
And also, how should copies for the last 2 weeks be stored in this case? Not 48 copies, but specifically one copy from two weeks ago and one from last week.

**Comment by Vikontrol** [comment]
this could be resolved if it were possible to include parameter count_instead_of_period  for daily, weekly, or monthly backups separately.


I'll make separate configurations for one database: 

1. one with the parameter count_instead_of_period enabled and with a schedule of:
```
    retention:
      count_instead_of_period: true
      days: 24
      weeks: 1
      months: 0
```
2. another without the count_instead_of_period parameter with a schedule of:
```
    retention:
      count_instead_of_period: false
      days: 0
      weeks: 2
      months: 0
```

**Comment by randreev1321** [comment]
Hi, @Vikontrol.

In this case, the solution you suggested is correct.

In case you want to keep a certain number of copies only for collecting daily copies, you should allocate a separate config for that using the count_instead_of_period option.

**Comment by Vikontrol** [comment]
Yes, but in this case, for the second config, you have to specify a different target directory, for example /var/nxs-backup/weekly/dump, otherwise the first config will delete the backups collected by the second one, and I will have 2 backups of the weekly, not 1 weekly and 1 bi-weekly.

Also, the problem is that when configuring 
```
    retention:
      count_instead_of_period: true
      days: 24
      weeks: 1
      months: 0
```
it on Sunday, I don't have 24 daily copies because the "weeks" parameter controls the number of copies on that day.

**Comment by randreev1321** [comment]
Hi, @Vikontrol.

Should to divide your question into 2 cases.

Option 1 - using a single job.

This variant is suitable if you collect backups more than once a day and you need only one weekly/monthly copy. Example, config:

```yaml 
    retention:
      count_instead_of_period: true
      days: 24
      weeks: 1
      months: 0
```

For this variant, the backup rotation on Sunday, when weekly copies are created and symlinks are created to them, has been fixed. 

Now, instead of deleting the weekly copies, they are moved to the symlinks location. This way you no longer lose your daily copies.

Option 2 - Using two jobs.

If you need to store more than one weekly/monthly copy, and you collect backups more than once a day, you should create separate tasks with different configurations.

The first config, for regular collection and rotation of copies once a day:

```yaml 
job_name: my_common_job 
... 
storages_options: 
  - storage_name: local 
    retention: 
      count_instead_of_period: false 
      days: 0 
      weeks: 4 
      months: 6 
```

Second config, to collect and rotate copies once an hour:

```yaml 
job_name: my_common_job_hourly 
... 
storages_options: 
  - storage_name: local 
    retention: 
      count_instead_of_period: true 
      days: 24 
      weeks: 0 
      months: 0 
```

With this set of configs, you can set `my_common_job_hourly` to run once an hour, and collect all jobs once a day.

Then you will get two catalogs with final copies, one will store copies of `my_common_job` and the other `my_common_job_hourly`. There is no need to configure different directories, as a subdirectory with the job name will be created in the target directory.

In case you create configs with the same job names, the copies will be collected in a common directory, and the rotation of periods that are configured to store 0 copies will be ignored. But in this case, we cannot guarantee that the `nxs-backup start my_job` command will select the job you expect to run.

---

The fixes will be available with the v3.6.1 release.

**Comment by randreev1321** [comment]
Your issue has been resolved in version [3.6.1](https://github.com/nixys/nxs-backup/releases/tag/v3.6.1).

Thanks for your interest in nxs-backup. We look forward to receiving new requests.
